### ISSUE TITLE
Implement query() and adding/removing indexes

### ISSUE THREAD

**Issue by nolanlawson** [pouchdb-community/relational-pouch#3]
This is another Ember Data thing that we don't support yet. (query(), anyway).

**Comment by oliviertassinari** [comment]
:+1:  It would be great.
I had to expose fromRawDoc() on the public API to make a query.

**Comment by nolanlawson** [comment]
Actually nowadays it may make more sense to just directly plug into `pouchdb-find` and expose that in `ember-pouch`. However, I don't have much time for that project these days.

**Comment by wujitouch** [comment]
I tried using pouchdb-find along side this plugin but since this one uses a "smart" ID for the document, pouchdb-find has trouble retrieving docs. I could have added the deserialize logic from this plugin but it turned out to be easier to work with the code without using both, so I just stuck with find.

**Comment by nolanlawson** [comment]
@wujitouch Yeah, I think pouchdb-find integration needs to be implemented within this plugin itself.

**Comment by jlami** [comment]
I added parseRelDocs in my PR #61 which I needed for my own find query. Thought it might be a good idea to expose this to the outside.

One extra problem i realised when looking at the filters was that db.find in pouchdb-find does not filter on types. So filtering on an attribute `date` might give more than one type back. But most of the time you only want one type back. For the `findHasMany` query I do this with an extra selector for the id.

So maybe it is better to make our own version of the query and index functions, since we might also want to include the _id between filter_ to select only documents of a certain type.
